File: Day 1/descending.cpp
#include<iostream>
#include<fstream>
#include<vector>

using namespace std;
typedef vector<int> List;

List windows(const List& L) {
    List M;
    for (int i = 0; i < L.size() - 2; ++i)
        M.push_back(L[i]+L[i+1]+L[i+2]);
    return M;
}

int descending(const List& L) {
    int sum = 0;
    for (int i = 0; i < L.size() - 1; ++i)
        sum += L[i+1] > L[i];
    return sum;
}

int main() {
    ifstream input ("input");
    List L;
    int n;
    while (input >> n)
        L.push_back(n);

    ofstream out;
    out.open ("output");
    out << descending(L) << '\n' << descending(windows(L)) << '\n';
    out.close();
    return 0;
}

File: Day 1/descending.py
M = [line.strip() for line in open('input')]
M = list(map(lambda x : int(x), M))
A = [1 for i in range(1, len(M)) if M[i] > M[i-1]]
B = [1 for i in range(0, len(M)-3) if 
      M[i+1] + M[i+2] + M[i+3] > M[i] + M[i+1] + M[i+2]
]
output = open('output', 'w')
output.write(str(sum(A))+'\n'+str(sum(B))+'\n')
output.close()